library/core/src/main/java/com/google/android/exoplayer2/mediacodec/MediaCodecRenderer.java

```diff
@@ -1132,6 +1133,7 @@ private void initCodec(MediaCodecInfo codecInfo, MediaCrypto crypto) throws Exce
       this.codecNeedsAdaptationWorkaroundBuffer =
           codecAdaptationWorkaroundMode != ADAPTATION_WORKAROUND_MODE_NEVER;
     }
+    codecNeedsSkipSurplusOutBufferWorkaround = codecNeedsSkipSurplusOutBufferWorkaround(codecName, codecInputFormat);
     if ("c2.android.mp3.decoder".equals(codecInfo.name)) {
```

Comment:
The format change to 1920x1080 occurs as an adaptive track selection.  Because the codec it self does not change.  So the logic will not trigger on a track change